push-image

```diff
@@ -1,28 +1,43 @@
-#!/bin/bash -eu
+#!/bin/bash
 
-# Push the 'cli:5' image to Dockerhub when on the 'master' branch
+set -e
 
-cd "$(git rev-parse --show-toplevel)"
+readonly REGISTRY="cyberark"
+readonly VERSION="$(cat VERSION)"
+readonly VERSION_TAG="5-${VERSION}"
 
-IMAGE='cyberark/conjur-cli'
+readonly IMAGES=(
+  "conjur-cli"
+)
 
-function tag_and_push() {
-    local image="$1"
-    local tag="$2"
-    local description="$3"
+readonly TAGS=(
+  "5"
+  "5-latest"
+  "$VERSION_TAG"
+)
 
-    echo "TAG = $tag, $description"
+# fetching tags is required for git_description to work
+git fetch --tags
+git_description=$(git describe)
 
-    docker tag "$image" "$image:$tag"
-    docker push "$image:$tag"
-}
+for image_name in "${IMAGES[@]}"; do
 
-version_tag="5-$(cat VERSION)"
+  # if it’s not a tagged commit, VERSION will have extra junk (i.e. -g666c4b2), so we won’t publish that commit
+  # only when tag matches the VERSION, push VERSION and latest releases
+  # and x and x.y releases
+  #Ex: 5-6.2.1
+  if [ "$git_description" = "v${VERSION_TAG}" ]; then
+    echo "Revision $git_description matches version $VERSION exactly. Pushing to Dockerhub..."
 
-tag_and_push $IMAGE '5'        'latest image'
-tag_and_push $IMAGE '5-latest'   'same as "5"'
-tag_and_push $IMAGE $version_tag 'version-specific image'
+    for tag in "${TAGS[@]}" $(gen_versions "$VERSION"); do
+      echo "Tagging and pushing $REGISTRY/$image_name:$tag"
 
-# push to legacy `conjurinc/cli5` tag
-docker tag "$IMAGE" conjurinc/cli5:latest
-docker push conjurinc/cli5:latest
+      docker tag "$image_name" "$REGISTRY/$image_name:$tag"
```

Comment:
The build script builds an image tagged `cyberark/conjur-cli:latest`, so this command should actually be
```suggestion
      docker tag "cyberark/conjur-cli:latest" "$REGISTRY/$image_name:$tag"
```

since this also has to be updated below, it'd be reasonable to store the value `cyberark/conjur-cli:latest` in a variable and update it in both places.